niworkflows/interfaces/freesurfer.py

```diff
@@ -326,6 +314,54 @@ def _run_interface(self, runtime):
         return runtime
 
 
+def fix_lta_length(lta_file):
+    """
+    Revises the length of the filename field in the LTA file and replaces if too long.
+    >>> import nibabel as nb
+    >>> import numpy as np
+    >>> import subprocess
+    >>> import os
+    >>> import shutil
+    >>> data = np.arange(4*4*3).reshape(4,4,3)
+    >>> long_name = os.getcwd() + '/Q/dc2/scratch/davhunt/workflows/5d6705bee0ae0618d1ededa8/5d6d\
+    ... b5f2e0ae06ba6cee3710/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_se\
+    ... s_SSSES_task_balloonanalogrisktask_run_1_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_bold_very_\
+    ... long_filename.nii.gz'
+    >>> short_name = 'short_filename.nii.gz'
+    >>> os.makedirs(os.path.dirname(long_name))
+    >>> nifti_file = nb.Nifti1Image(data, affine=np.eye(4))
+    >>> nb.save(nifti_file, long_name)
+    >>> nb.save(nifti_file, short_name)
+    >>> devnull = open(os.devnull, 'w')
+    >>> subprocess.call(["lta_convert", "--inlta", "identity.nofile", "--src", long_name, "--trg"
+    ... , long_name, "--outlta", "lta_file_1.lta"], stdout=devnull)
+    0
+    >>> subprocess.call(["lta_convert", "--inlta", "identity.nofile", "--src", short_name,
+    ... "--trg", short_name, "--outlta", "lta_file_2.lta"], stdout=devnull)
+    0
+    >>> fix_lta_length('lta_file_1.lta')
+    True
+    >>> fix_lta_length('lta_file_2.lta')
+    False
+    >>> shutil.rmtree('/Q')
+    >>> os.remove(short_path); os.remove('lta_file_1.lta'); os.remove('lta_file_2.lta')
```

Comment:
The code looks fine, but all of this setup should not go in a doctest.

Just add the two LTA files in https://github.com/poldracklab/niworkflows/tree/master/niworkflows/tests/data, perhaps with names like `valid_transform.lta` and `long_path_transform.lta`.

Then you'll need to make the files available from a doctest here:

https://github.com/poldracklab/niworkflows/blob/c4ceb2ea570a30a406ea50c0d97a7374e596f6b8/niworkflows/conftest.py#L17-L33

I would use:

```Python
doctest_namespace['test_data'] = pkg_resources.resource_filename('niworkflows', 'tests/data')
```

Then your doctest can become something like:

```
""" Fix the length of the filename field in an LTA file if too long.

Updates file in place.

Examples
--------

No changes are made to a valid transform file:

>>> valid_transform = Path(test_data) / 'valid_transform.lta'
>>> orig_contents = valid_transform.read_text()
>>> fix_lta_length(valid_transform)
False
>>> valid_transform.read_text() == orig_contents
True

Invalid transform files are files with > 255 characters in any line:

>>> invalid_transform = Path(test_data) / 'long_path_transform.lta'
>>> orig_contents = invalid_transform.readtext()
>>> any(len(line) > 255 for line in orig_contents.splitlines(keepends=True))
True
>>> local_copy = Path('invalid.lta')
>>> local_copy.write_text(orig_contents)
>>> fix_lta_length(local_copy)
True
>>> local_copy.read_text() == orig_contents
False
"""
```